Sketch.java

```diff
@@ -1,36 +1,453 @@
+      //moonBlock = 1
+      //empty space = 0
+      //meteor = 4
+      //coloured rectangles = 6
+      {1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1},
+      {1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1},
+      {1, 0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1},
+      {1, 0, 1, 0, 0, 1, 0, 1, 6, 0, 0, 0, 0, 6, 1, 0, 0, 0, 1, 1},
+      {1, 0, 1, 0, 0, 1, 0, 1, 6, 0, 0, 0, 0, 6, 1, 0, 0, 0, 1, 1},
+      {1, 0, 1, 0, 0, 1, 0, 1, 1, 6, 0, 0, 6, 1, 1, 0, 0, 0, 1, 1},
+      {1, 0, 1, 0, 0, 1, 0, 1, 6, 1, 6, 6, 1, 6, 1, 0, 0, 0, 1, 1},
+      {1, 0, 1, 0, 0, 1, 0, 1, 4, 1, 1, 1, 1, 4, 1, 0, 0, 0, 1, 1},
+      {1, 0, 1, 1, 1, 1, 0, 1, 6, 1, 6, 6, 1, 6, 1, 0, 0, 0, 1, 1},
+      {1, 0, 0, 0, 0, 0, 0, 1, 4, 1, 4, 4, 1, 4, 1, 0, 0, 3, 1, 1},
+      {1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1},
+    };
+   
+   for(int row = 0; row < levelData.length; row++) {
+     for(int column = 0; column < levelData [row].length; column++) {
+       int cell = levelData[row] [column];
+
+       // using the numbers on the 2D array to determine where each item is placed
+       if(cell == 1) {
+         image(moonBlock, column * cellSize, row * 40);
+        }
+
+        if (cell == 3){
+          PImage earth = loadImage("Pics/BigEarth.png");
+          earth.resize(40, 40);
+          image(earth, (column * 40), (row * 40));
+        } 
+
+        if(cell == 4) {
+          PImage meteor = loadImage("Pics/comet.png");
+          meteor.resize(40, 40);
+          image(meteor, (column * 40), (row * 40));
+        }
+
+        if(cell == 6) {
+          noStroke();
+          fill(250, 123, 12);
+          rect((column * 40), (row * 40), 40, 40);
+        }
+      }
+    }
+  } 
+
+  /**
+   * Draws the game map
+   * 
+   * @param cellSize size of each 2-D arrays cell
+   */
+
+  public void keyPressed() {
+
+    int speed = 40;    
+   
+    // user input & interactivity
+    // rocket moves accordingly to what arrow directions user presses
+    if (keyPressed) {
+      
+      if (keyCode == UP) {
+      
+        if(imgRocketY < 80){
+          speed = 0;
+        }
+        else if(imgRocketY == 360){
+          speed = 0;
+          if (imgRocketX == 240){
+            speed = 40;
+            imgRocketY -= speed;
+          }
+        }
+        else{
+          speed = 40;
+          imgRocketY -= speed;
+        }
+      }
+       
+      else if (keyCode == DOWN) {
+
+        if(imgRocketY > 320){
+           speed = 0;
+        }
+        else{
+          speed = 40;
+          imgRocketY += speed;
+        }
+      } 
+
+      else if(keyCode == LEFT){
+
+       if(imgRocketX < 80){
+         speed = 0;
+        }
+        else if(imgRocketX == 240){
+          speed = 0;
+        }
+        else if(imgRocketX == 600){
+          speed = 0;
+        }
+        else{
+          imgRocketX -= speed;
+        }
+      }
+
+      else if(keyCode == RIGHT){
+    
+        if(imgRocketX > 660){
```

Comment:
looks like the collision is hardcoded in based on simple x co-ordinates.  A missed opportunity to leverage the 2D array to manage your collision checks.